lit_nlp/client/modules/perturbations_table_module.ts

```diff
@@ -0,0 +1,310 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// tslint:disable:no-new-decorators
+import '@material/mwc-icon';
+
+import * as d3 from 'd3';
+import {customElement, html} from 'lit-element';
+import {computed, observable} from 'mobx';
+import {classMap} from 'lit-html/directives/class-map';
+import {styleMap} from 'lit-html/directives/style-map';
+
+import '../elements/text_diff';
+import {app} from '../core/lit_app';
+import {LitModule} from '../core/lit_module';
+import {TableData} from '../elements/table';
+import {CallConfig, FacetMap, GroupedExamples, IndexedInput, LitName, ModelsMap, Spec} from '../lib/types';
+import {doesOutputSpecContain, formatLabelNumber, findSpecKeys} from '../lib/utils';
+import {GroupService} from '../services/group_service';
+import {DeltasService, RegressionService, ClassificationService, SliceService} from '../services/services';
+import {RegressionInfo} from '../services/regression_service';
+import {DeltaRow, DeltaInfo, Source} from '../services/deltas_service';
+
+import {styles} from './perturbations_table_module.css';
+import {styles as sharedStyles} from './shared_styles.css';
+
+type DeltaRowsById = {
+  [id: string]: DeltaRow
+};
+
+/**
+ * Module to sort generated countefactuals by the change in prediction for a
+ regression or multiclass classification model.
+ */
+@customElement('perturbations-table-module')
+export class PerturbationsTableModule extends LitModule {
+  static title = 'Perturbations';
+  static numCols = 4;
+  static duplicateForModelComparison = true;
+  static duplicateAsRow = false;
+  static template = (model = '') => {
+    return html`<perturbations-table-module model=${model}></perturbations-table-module>`;
+  };
+
+  static shouldDisplayModule(modelSpecs: ModelsMap, datasetSpec: Spec) {
+    return doesOutputSpecContain(modelSpecs, [
+      'RegressionScore',
+      'MulticlassPreds'
+    ]);
+  }
+
+  static get styles() {
+    return [sharedStyles, styles];
+  }
+
+  private readonly regressionService = app.getService(RegressionService);
+  private readonly classificationService = app.getService(ClassificationService);
+  private readonly deltasService = app.getService(DeltasService);
+
+  /* These constants are for referring to table columns */
+  private readonly ID_COLUMN = 0;
+  private readonly SENTENCE_COLUMN = 1;
+  private readonly DELTA_COLUMN = 5;
+
+  @observable private filterSelected = true;
+  @observable private lastSelectedSourceIndex?: number;
+
+  private onSelect(selectedRowIndices: number[]) {
+    const ids = selectedRowIndices
+                    .map(index => this.appState.currentInputData[index]?.id)
+                    .filter(id => id != null);
+    this.selectionService.selectIds(ids);
+  }
+
+  private onPrimarySelect(index: number) {
```

Comment:
not convinced that this module should only show selected datapoints and their related points. to me it seems more natural to display all generated counterfactuals

Reply:
@jameswex I left the checkbox, it defaults to unchecked.  I could remove from this PR if you want, but I'd vote to keep it in since in the next PR it allows brushing in the vis and seeing what you brushed in the table.